### PR TITLE
[fixes] handle exceptions

### PR DESCRIPTION
**Pull request by saurabh6790**
- Payment Required exception

### CODE REVIEW

File: erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py [hunk 1]
```diff
@@ -67,7 +68,12 @@ def sync_shopify():
 			update_item_stock_qty()
 
 		except ShopifyError:
-			frappe.db.set_value("Shopify Settings", None, "enable_shopify", 0)
+			disable_shopify()
+			
+		except requests.exceptions.HTTPError, e:
+			if e.args[0] and e.args[0].startswith("402"):
+				disable_shopify()
+				sendmail_to_intimate_sync_disabled(_(""" Your shopify account has been disabled. Payment Required."""))
```
Comment (anandpdoshi) [inline erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py]: - No need to have a function `sendmail_to_intimate_sync_disabled`.
- Also, there's an empty space at the start in the message.
- `_("Shopify has suspended your account till you complete the payment. We have disabled ERPNext Shopify Sync. Please enable it once your complete the payment at Shopify.")`

File: erpnext_shopify/utils.py [hunk 2]
```diff
@@ -167,9 +167,21 @@ def create_webhooks():
...
 	"""Disable Item if not exist on shopify"""
+	frappe.db.rollback()
 	item.sync_with_shopify = 0
 	item.save()
 	frappe.db.commit()
 
 def get_shopify_item_image(shopify_id):
 	return get_request("/admin/products/{0}/images.json".format(shopify_id))["images"]
+
+def disable_shopify():
```
Comment (anandpdoshi) [inline erpnext_shopify/utils.py]: misleading function names `disable_shopify_sync` and `disable_shopify`
Better names:
- `disable_shopify_sync_for_item`
- `disable_shopify_sync_on_exception`

File: erpnext_shopify/utils.py [hunk 3]
```diff
@@ -167,9 +167,21 @@ def create_webhooks():
...
+	frappe.db.rollback()
+	frappe.db.set_value("Shopify Settings", None, "enable_shopify", 0)
+	frappe.db.commit()
+
+def sendmail_to_intimate_sync_disabled(message):
+	frappe.sendmail(get_system_manager(), subject=_("Shopify Sync has been disabled"), content=message)
+
+def get_system_manager():
+	return frappe.db.sql("select parent from tabUserRole where role='System Manager' and parent != 'Administrator'")[0][0]
```
Comment (anandpdoshi) [inline erpnext_shopify/utils.py]:
```
from frappe.utils.user import get_system_managers
get_system_managers()
```

File: erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py [hunk 4]
```diff
@@ -67,7 +68,12 @@ def sync_shopify():
 			update_item_stock_qty()
 
 		except ShopifyError:
-			frappe.db.set_value("Shopify Settings", None, "enable_shopify", 0)
+			disable_shopify()
+			
+		except requests.exceptions.HTTPError, e:
+			if e.args[0] and e.args[0].startswith("402"):
```
Comment (anandpdoshi) [inline erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py:74]: add the complete error message as comment over here as a NOTE

**Comment by saurabh6790** [comment]
@anandpdoshi Fixed.

File: erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py [hunk 5]
```diff
@@ -543,9 +550,14 @@ def sync_shopify_orders():
...
 def validate_customer_and_product(order):
-	if not frappe.db.get_value("Customer", {"shopify_id": order.get("customer").get("id")}, "name"):
-		create_customer(order.get("customer"))
-
+	customer_id = order.get("customer", {}).get("id")
+	if not customer_id:
+		if not frappe.db.get_value("Customer", {"shopify_id": customer_id}, "name"):
+			create_customer(order.get("customer"))
+	else:
+		frappe.msgprint(__("Customer is mandatory to create order"))
```
Comment (anandpdoshi) [inline erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py:558]: This condition looks so strange.

The else part looks like `if customer_id` show message "Customer is mandatory". How does this work out?

**Pull request closed without merging**